.github/workflows/sycl_linux_build_and_test.yml

```diff
@@ -111,6 +111,7 @@ jobs:
       run: cmake --build $GITHUB_WORKSPACE/build --target deploy-sycl-toolchain
     # TODO allow to optionally disable in-tree checks
     - name: check-mlir-sycl
+      shell: bash
```

Comment:
Why add `shell: bash`? I see that the `sycl_linux_build_and_test.yml` in `sycl` branch https://github.com/intel/llvm/blob/sycl/.github/workflows/sycl_linux_build_and_test.yml doesn't do that.

Reply:
Because we are already using it for `check-cgeist` at line 119.

Reply:
Is there a reason you prefer to remove the shell ? I can go either way as long as we are consistent.

Reply:
to be consistent with sycl branch.

Reply:
Looks like we intentionally added it in https://github.com/intel/llvm/pull/7085, let's keep it then.

Reply:
On further thoughts I am splitting the PR. This one will deal with moving the analysis files. A future one will deal with adding the polygeist unittests to the build scripts.